accounts/views.py

```diff
@@ -143,4 +94,33 @@ def test_token(request):
         json: validation msg
     """
     print(request.user)
-    return Response("passed for {}".format(request.user.username))
\ No newline at end of file
+    return Response("passed for {}".format(request.user.username))
+
+@api_view(['GET'])
+def get_user_by_username(request, username):
+    try:
+        user = User.objects.get(username=username)
```

Comment:
I mean here I start to be lost a bit as well :D ....so I need to think about all the logic more.....

Scenarios: 
get_user_public_profile ---> need to add condition if is_visible. ---> so that will send data about user. To anyone who will request that via "more info" button on ther main page. Just note (If visitor can see the user profile on the main page that means that user have is_visible=True...but it will be better to implement it in view itself)

get_user_personal_profile ---> need to add condition reqest.user == auth.user. So that means basicly that User request his onw data so he wanna see his profile.

get_visible_users ----> clear enought i guess.

Reply:
When you write down the scenarios it makes sense :) I recommend writing this kind of description (scenarios, use cases etc.) in the issue and pull request description. It'll make us stay on the track.

As @trnecka agreed, I'd also go with the name that clearly says we get the logged user profile here. I imagine a button "My Profile", or something like that, that leads to my profile I can edit. It's the place where the `get_user_personal_profile` is used.

Reply:
So....I've changed name of view from get_user_public_profile to get_logged_user_profile add conditions is_visible and request.user.username != username